src/plugins/workspace/server/plugin.ts

```diff
@@ -157,7 +158,13 @@ export class WorkspacePlugin implements Plugin<WorkspacePluginSetup, WorkspacePl
         enabled: true,
         permissionEnabled: isPermissionControlEnabled,
       },
+      dashboards: { isDashboardAdmin: true },
     }));
+    // Dynamically update capabilities based on the auth information from request.
+    core.capabilities.registerSwitcher((request) => {
+      const isDashboardAdmin = !!getWorkspaceState(request).isDashboardAdmin;
```

Comment:
When `isPermissionControlEnabled = false` and `this.setupPermission(core);` is not run. In such case, `getWorkspaceState(request).isDashboardAdmin` is `undefined`, then it will return `{ dashboards: { isDashboardAdmin: false } };`, but when permission control is disabled, `isDashboardAdmin` should be `true`. Am I right?

Reply:
Yes, the code has been updated